media-sound/jack: fix #298850, new EAPI and eclass

madmartin opened the pull request:
- fix bug https://bugs.gentoo.org/show_bug.cgi?id=298850
- EAPI 6
- remove eclass distutils in favor of python-single-r1
- add functional enhancement: add cddb server freedb.musicbrainz.org

Package-Manager: Portage-2.3.3, Repoman-2.3.1

aballier commented:
@mgorny

mgorny commented on media-sound/jack/files/jack-3.1.1-python26.patch:
```diff
@@ -0,0 +1,12 @@
+--- a/jack_functions.py
```
Why are you adding a second copy of the patch that is already there?

mgorny replied:
Ok, that explains it. You can either modify the old file, i.e. change the header there, or just remove it along with the old ebuild.

mgorny commented on media-sound/jack/files/shebang-python27.patch:
```diff
@@ -0,0 +1,8 @@
+--- a/jack	2016-11-16 00:34:18.402706654 +0100
```
The eclasses have `python_fix_shebang` function for that.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,50 @@
+
+PYTHON_COMPAT=( python2_7 )
+PYTHON_REQ_USE="ncurses"
+inherit python-single-r1
```
Why are you using python-single-r1 instead of distutils-r1? You are patching setup.py but not using it, and reinventing the whole install.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,50 @@
+
+DEPEND="${PYTHON_DEPS}"
+RDEPEND="${DEPEND}
+	dev-python/cddb-py
```
Every dependency with python modules needs `[${PYTHON_USEDEP}]`.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,50 @@
+	media-sound/lame
+	media-sound/cdparanoia"
+REQUIRED_USE=${PYTHON_REQUIRED_USE}
+PATCHES="${FILESDIR}/${PN}-3.1.1-python26.patch
```
PATCHES should be a bash array, not text var.

mgorny commented on media-sound/jack/files/add-cddbserver-musicbrainz.patch:
```diff
@@ -0,0 +1,15 @@
+--- a/jack_freedb.py	2010-07-16 23:37:43.304254370 +0200
```
Please use consistent patch naming, i.e. either make them all start with `jack-VERSION-...`, or none.

SoapGentoo replied:
always prefer `a/` `b/` (git-style) as this is version agnostic

mgorny replied:
I was talking about filename, not what's inside…

SoapGentoo commented on media-sound/jack/files/shebang-python27.patch:
```diff
@@ -0,0 +1,8 @@
++++ b/jack	2016-11-16 00:34:09.994636061 +0100
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python
++#!/usr/bin/python2.7
```
also, you're making it worse, the proper shebang should be
`#!/usr/bin/env python2`

mgorny commented:
@madmartin, the most important point is that you should really use distutils-r1 unless you have a very good reason not to. This replaces most of the install phase.

Also, please add yourself to metadata.xml and the proxy-maint project.

madmartin commented:
@mgorny, the requested changes are applied. I hope you like it...

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+DEPEND="${PYTHON_DEPS}"
+RDEPEND="${DEPEND}
+	dev-python/cddb-py[${PYTHON_USEDEP}]
```
Please check whether the package builds successfully without those Python modules installed (i.e. unmerge it, then `emerge -1v --nodep media-sound/jack`. If not, they should land in DEPEND too.

madmartin replied:
`emerge -1v --nodep media-sound/jack` works without modules installed, so placing them in RDEPEND is fine.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+EAPI=6
+
+DISTUTILS_SINGLE_IMPL=1
```
Remove this. It's only needed because you are using the eclass wrong.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	"${FILESDIR}/${P}-add-cddbserver-musicbrainz.patch"
+)
+
+src_install() {
```
Replace it with `python_install()` and `python_install_all()` appropriately. In the former, call `distutils-r1_python_install` which runs distutils and python_doscript. In the latter, set DOCS/HTML_DOCS, call `distutils-r1_python_install_all` and install all remaining files.

mgorny commented on media-sound/jack/metadata.xml:
```diff
@@ -5,4 +5,12 @@
 	<email>sound@gentoo.org</email>
 	<name>Gentoo Sound project</name>
 </maintainer>
+<maintainer type="person">
```
You go first.

madmartin commented:
@mgorny, `DISTUTILS_SINGLE_IMPL=1' was necessary because /usr/bin/jack was not installed properly without it. I read python setup.py documentation and modified setup.py once more, added `scripts=['jack']`. Now `python_install` works fine without `DISTUTILS_SINGLE_IMPL=...'

madmartin commented:
@mgorny ping....

mgorny commented on media-sound/jack/files/jack-3.1.1-dontuse-jack-cursesmodule.patch:
```diff
@@ -0,0 +1,59 @@
+--- a/jack_t_curses.py	2004-12-12 05:54:36.000000000 +0100
```
Since the patch is no longer just about jack_curses, I suppose you'd want to rename it.

mgorny commented on media-sound/jack/files/jack-3.1.1-dontuse-jack-cursesmodule.patch:
```diff
@@ -0,0 +1,59 @@
++    # we dont need jack_cursemodule any more because its included in python 
++    # since release 2.0
++    # see http://docs.python.org/whatsnew/2.0.html
++    info("jack_curses module not found, using normal curses...")
```
You probably don't want to print that. What's the point, if we are never installing it?

mgorny commented on media-sound/jack/files/jack-3.1.1-dontuse-jack-cursesmodule.patch:
```diff
@@ -0,0 +1,59 @@
+-        def resizeterm(y, x):
+-            pass
++        from curses import endwin, resizeterm, A_REVERSE, newwin, newpad, initscr, noecho, cbreak, echo, nocbreak
++        # def resizeterm(y, x):
```
Do not readd commented-out stuff. There's no point for that and it only increases patch size.

madmartin commented:
@mgorny The Postman Rings Again...

madmartin commented:
@mgorny would you take a look here please?

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	"${FILESDIR}/${P}-add-cddbserver-musicbrainz.patch"
+)
+
+python_install() {
```
This function is the same as the default, so you can remove it.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	insinto /etc
+	newins example.etc.jackrc jackrc
+	newman jack.man jack.1
+	DOCS=( README doc/ChangeLog doc/TODO )
```
Please mark those variables `local` since they aren't needed after leaving the function.

mgorny commented on media-sound/jack/jack-3.1.1-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	newins example.etc.jackrc jackrc
+	newman jack.man jack.1
+	DOCS=( README doc/ChangeLog doc/TODO )
+	HTML_DOCS=( doc/*html doc/*css doc/*gif )
```
A minor nitpick but I think `*.html` etc. (i.e. with a dot) would be more readable.
